_signal_processing/krakenSDR_signal_processor.py

```diff
@@ -283,6 +291,48 @@ def find_nearest(array, value):
                 measured_spec_mean = np.mean(measured_spec[vfo_start_measure_spec:vfo_end_measure_spec])
                 self.vfo_squelch[i] = measured_spec_mean + self.default_auto_channel_db_offset
 
+    def save_processing_status(self) -> None:
+        """This method serializes system status to file."""
+
+        status = {}
+        daq_status = {}
+
+        status["timestamp_ms"] = int(time.time() * 1e3)
+        status["station_id"] = self.station_id
+        status["hardware_id"] = self.module_receiver.iq_header.hardware_id.rstrip("\x00")
+        status["unit_id"] = self.module_receiver.iq_header.unit_id
+        status["host_os_type"] = SYSTEM_UNAME.system
+        status["host_os_version"] = SYSTEM_UNAME.release
+        status["host_os_architecture"] = SYSTEM_UNAME.machine
+        status["software_version"] = SOFTWARE_VERSION
+        status["software_git_hash"] = SOFTWARE_GIT_HASH
+        status["uptime_ms"] = int(time.monotonic() * 1e3)
+        status["gps_status"] = self.gps_status
+        status["daq_connected"] = self.module_receiver.receiver_connection_status
+        if status["daq_connected"]:
+            status["timestamp_ms"] = self.module_receiver.iq_header.time_stamp
+            daq_status["data_frame_index"] = self.module_receiver.iq_header.cpi_index
+            daq_status["frame_sync"] = not bool(self.module_receiver.iq_header.check_sync_word())
+            daq_status["sample_delay_sync"] = bool(self.module_receiver.iq_header.delay_sync_flag)
+            daq_status["iq_sync"] = bool(self.module_receiver.iq_header.iq_sync_flag)
+            daq_status["noise_source_enabled"] = bool(self.module_receiver.iq_header.noise_source_state)
+            daq_status["adc_overdrive"] = bool(self.module_receiver.iq_header.adc_overdrive_flags)
+            daq_status["sampling_frequency_hz"] = self.module_receiver.iq_header.adc_sampling_freq
+            daq_status["bandwidth_hz"] = self.module_receiver.iq_header.sampling_freq
+            daq_status["decimated_bandwidth_hz"] = self.module_receiver.iq_header.sampling_freq // self.dsp_decimation
+            daq_status["buffer_size_ms"] = (
+                self.module_receiver.iq_header.cpi_length / self.module_receiver.iq_header.sampling_freq
+            ) * 1e3
+            daq_status["dropped_frames"] = self.dropped_frames
+
```

Comment:
Might make sense adding a general status. By checking which you can see whether the receiver is ready or not.

Reply:
Fair enough, added `daq_ok`.